p2p/discovery/refresher.go

```diff
@@ -27,29 +27,28 @@ var ErrBootAbort = errors.New("bootstrap canceled by signal")
 
 // refresher is used to bootstrap and requestAddresses peers in the addrbook
 type refresher struct {
-	logger log.Log
-	config config.SwarmConfig
+	logger       log.Log
+	localAddress *node.NodeInfo
 
+	book      addressBook
 	bootNodes []*node.NodeInfo
 
-	book *addrBook
-
+	disc        Protocol
 	lastQueries map[p2pcrypto.PublicKey]time.Time
 
-	disc Protocol
-
 	quit chan struct{}
 }
 
-func newRefresher(book *addrBook, disc Protocol, bootnodes []*node.NodeInfo, logger log.Log) *refresher {
+func newRefresher(local *node.NodeInfo, book addressBook, disc Protocol, bootnodes []*node.NodeInfo, logger log.Log) *refresher {
 	//todo: trigger requestAddresses every X with random nodes
 	return &refresher{
-		logger:      logger,
-		book:        book,
-		disc:        disc,
-		bootNodes:   bootnodes,
-		lastQueries: make(map[p2pcrypto.PublicKey]time.Time),
-		quit:        make(chan struct{}), // todo: context ?
+		logger:       logger,
+		localAddress: local,
+		book:         book,
+		disc:         disc,
+		bootNodes:    bootnodes,
+		lastQueries:  make(map[p2pcrypto.PublicKey]time.Time),
+		quit:         make(chan struct{}), // todo: context ?
 	}
 }
 
```

Comment:
rename `minPeers`

Reply:
Also, is it possible that Bootstrap will be executed from two routines in parallel?

Reply:
refresher isn't threadsafe right now. Bootstrap is called once. (though we don't have any blocker to call it twice). this is the reason `refresh` is an unexported method. All the peers selection process happens one at a time though it enforces refreshes to happen one at a time.
We'll have to consider thread-safety if we allow calling refresh from different threads. or maybe just don't allow bootstrap to refresh on a second call. but we'll tackle it right away since everything will explode from concurrent access to unprotected resource.